src/sentry/migrations/0276_backfill_dashboard_widget_query_columns_aggregates.py

```diff
@@ -0,0 +1,82 @@
+# Generated by Django 2.2.24 on 2022-03-02 21:21
+
+import re
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+AGGREGATE_PATTERN = r"^(\w+)\((.*)?\)$"
+AGGREGATE_BASE = r".*(\w+)\((.*)?\)"
+EQUATION_PREFIX = "equation|"
+
+
+def is_equation(field: str) -> bool:
+    """check if a public alias is an equation, which start with the equation prefix
+    eg. `equation|5 + 5`
+    """
+    return field.startswith(EQUATION_PREFIX)
+
+
+def is_aggregate(field: str) -> bool:
+    field_match = re.match(AGGREGATE_PATTERN, field)
+    if field_match:
+        return True
+
+    equation_match = re.match(AGGREGATE_BASE, field)
+    if equation_match and is_equation(field):
+        return True
+
+    return False
+
+
+def backfill_columns_aggregates(apps, schema_editor):
+    DashboardWidgetQuery = apps.get_model("sentry", "DashboardWidgetQuery")
+
+    for widget_query in RangeQuerySetWrapperWithProgressBar(DashboardWidgetQuery.objects.all()):
+        if widget_query.columns or widget_query.aggregates:
+            continue
+
+        fields = widget_query.fields or []
+        columns = []
+        aggregates = []
+
+        for field in fields:
+            if is_aggregate(field):
+                aggregates.append(field)
+            else:
+                columns.append(field)
+
```

Comment:
Do we need to set `widget_query.columns = columns` here?

Reply:
Also need to do with `aggregates`. Could we just operate on the values in `widget_query` directly?

Reply:
oh wow rip